app/controllers/ems_common.rb

```diff
@@ -319,18 +319,24 @@ def update_button_validate
   end
   private :update_button_validate
 
-  def validate_credentials(verify_ems)
+  def get_validation_object(verify_ems)
```

Comment:
I understand why this is named `get_` but this is not done in Ruby. The call wants to be a in a separate class it seems to me. Probably this bit plus the method below at least + a small spec around that should be written.